[GPU Process] [FormControls] Rename ControlPart to ControlPartType and make it enum class

shallawa opened the pull request:
#### c9a83d0bf5725764c9064eb91af724c3df4a3495
<pre>
[GPU Process] [FormControls] Rename ControlPart to ControlPartType and make it enum class
<a href="https://bugs.webkit.org/show_bug.cgi?id=249167">https://bugs.webkit.org/show_bug.cgi?id=249167</a>
rdar://103266751

Reviewed by Aditya Keerthi.

The plan is to add a new class called ControlPart in which arguments from the DOM
elements and the render elements will be passed to GraphicsContext. GraphicsContext
will be responsible of drawing the ControlPart so we can delegate the drawing as
a DisplayList item to GPUP.

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
(WebCore::AccessibilityRenderObject::isApplePayButton const):
* Source/WebCore/accessibility/AccessibilitySlider.cpp:
(WebCore::AccessibilitySlider::orientation const):
* Source/WebCore/css/CSSPrimitiveValueMappings.h:
(WebCore::CSSPrimitiveValue::CSSPrimitiveValue):
(WebCore::CSSPrimitiveValue::operator ControlPartType const):
(WebCore::CSSPrimitiveValue::operator ControlPart const): Deleted.
* Source/WebCore/css/CSSValueKeywords.in:
* Source/WebCore/html/RangeInputType.cpp:
(WebCore::RangeInputType::handleKeydownEvent):
* Source/WebCore/html/shadow/SliderThumbElement.cpp:
(WebCore::hasVerticalAppearance):
(WebCore::SliderThumbElement::resolveCustomStyle):
* Source/WebCore/html/shadow/TextControlInnerElements.cpp:
(WebCore::SearchFieldResultsButtonElement::resolveCustomStyle):
(WebCore::SearchFieldCancelButtonElement::resolveCustomStyle):
* Source/WebCore/platform/Theme.cpp:
(WebCore::Theme::baselinePositionAdjustment const):
(WebCore::Theme::controlFont const):
(WebCore::Theme::controlSize const):
(WebCore::Theme::minimumControlSize const):
(WebCore::Theme::controlRequiresPreWhiteSpace const):
(WebCore::Theme::paint):
(WebCore::Theme::inflateControlPaintRect const):
(WebCore::Theme::controlBorder const):
(WebCore::Theme::controlPadding const):
* Source/WebCore/platform/Theme.h:
* Source/WebCore/platform/ThemeTypes.cpp:
* Source/WebCore/platform/ThemeTypes.h:
* Source/WebCore/platform/adwaita/ThemeAdwaita.cpp:
(WebCore::ThemeAdwaita::controlSize const):
(WebCore::ThemeAdwaita::minimumControlSize const):
(WebCore::ThemeAdwaita::controlBorder const):
(WebCore::ThemeAdwaita::paint):
* Source/WebCore/platform/adwaita/ThemeAdwaita.h:
* Source/WebCore/platform/graphics/formcontrols/ControlPartType.cpp: Added.
(WebCore::operator&lt;&lt;):
* Source/WebCore/platform/graphics/formcontrols/ControlPartType.h: Added.
* Source/WebCore/platform/mac/ThemeMac.h:
* Source/WebCore/platform/mac/ThemeMac.mm:
(WebCore::configureToggleButton):
(WebCore::createToggleButtonCell):
(WebCore::sharedRadioCell):
(WebCore::sharedCheckboxCell):
(WebCore::paintToggleButton):
(WebCore::setUpButtonCell):
(WebCore::button):
(WebCore::paintButton):
(WebCore::paintColorWell):
(WebCore::ThemeMac::baselinePositionAdjustment const):
(WebCore::ThemeMac::controlFont const):
(WebCore::ThemeMac::controlSize const):
(WebCore::ThemeMac::minimumControlSize const):
(WebCore::ThemeMac::controlBorder const):
(WebCore::ThemeMac::controlPadding const):
(WebCore::ThemeMac::inflateControlPaintRect const):
(WebCore::ThemeMac::paint):
* Source/WebCore/rendering/RenderAttachment.cpp:
(WebCore::RenderAttachment::shouldDrawBorder const):
* Source/WebCore/rendering/RenderLayerScrollableArea.cpp:
(WebCore::RenderLayerScrollableArea::updateScrollbarsAfterStyleChange):
(WebCore::RenderLayerScrollableArea::updateScrollbarsAfterLayout):
* Source/WebCore/rendering/RenderListBox.cpp:
(WebCore::RenderListBox::createScrollbar):
* Source/WebCore/rendering/RenderMenuList.cpp:
(RenderMenuList::menuStyle const):
(RenderMenuList::clientPaddingLeft const):
(RenderMenuList::clientPaddingRight const):
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::adjustAppearanceForElement const):
(WebCore::isAppearanceAllowedForAllElements):
(WebCore::RenderTheme::adjustStyle):
(WebCore::RenderTheme::autoAppearanceForElement const):
(WebCore::RenderTheme::paint):
(WebCore::RenderTheme::paintBorderOnly):
(WebCore::RenderTheme::paintDecorations):
(WebCore::RenderTheme::isControlContainer const):
(WebCore::RenderTheme::isControlStyled const):
(WebCore::RenderTheme::supportsFocusRing const):
(WebCore::RenderTheme::isDefault const):
(WebCore::RenderTheme::supportsMeter const):
(WebCore::RenderTheme::paintSliderTicks):
* Source/WebCore/rendering/RenderTheme.h:
(WebCore::RenderTheme::scrollbarControlSizeForPart):
(WebCore::RenderTheme::canPaint const):
* Source/WebCore/rendering/RenderThemeAdwaita.cpp:
(WebCore::RenderThemeAdwaita::supportsFocusRing const):
(WebCore::RenderThemeAdwaita::popupInternalPaddingBox const):
(WebCore::RenderThemeAdwaita::paintMenuList):
(WebCore::RenderThemeAdwaita::paintSliderTrack):
(WebCore::RenderThemeAdwaita::adjustSliderThumbSize const):
(WebCore::RenderThemeAdwaita::paintSliderThumb):
* Source/WebCore/rendering/RenderThemeIOS.h:
* Source/WebCore/rendering/RenderThemeIOS.mm:
(WebCore::RenderThemeIOS::canPaint const):
(WebCore::RenderThemeIOS::adjustedPaintRect const):
(WebCore::RenderThemeIOS::baselinePosition const):
(WebCore::RenderThemeIOS::isControlStyled const):
(WebCore::RenderThemeIOS::popupInternalPaddingBox const):
(WebCore::canAdjustBorderRadiusForAppearance):
(WebCore::RenderThemeIOS::paintSliderTrack):
(WebCore::RenderThemeIOS::adjustSliderThumbSize const):
(WebCore::RenderThemeIOS::adjustButtonStyle const):
(WebCore::RenderThemeIOS::supportsBoxShadow const):
(WebCore::RenderThemeIOS::supportsMeter const):
(WebCore::RenderThemeIOS::paintSliderTicks):
(WebCore::RenderThemeIOS::paintSliderTrackWithFormControlRefresh):
* Source/WebCore/rendering/RenderThemeMac.h:
* Source/WebCore/rendering/RenderThemeMac.mm:
(WebCore::RenderThemeMac::canPaint const):
(WebCore::RenderThemeMac::isControlStyled const):
(WebCore::RenderThemeMac::adjustRepaintRect):
(WebCore::RenderThemeMac::controlSupportsTints const):
(WebCore::RenderThemeMac::meterSizeForBounds const):
(WebCore::RenderThemeMac::supportsMeter const):
(WebCore::RenderThemeMac::levelIndicatorFor const):
(WebCore::RenderThemeMac::progressBarRectForBounds const):
(WebCore::RenderThemeMac::popupInternalPaddingBox const):
(WebCore::RenderThemeMac::paintSliderTrack):
(WebCore::RenderThemeMac::paintSliderThumb):
(WebCore::RenderThemeMac::adjustSliderThumbSize const):
* Source/WebCore/rendering/RenderThemeWin.cpp:
(WebCore::RenderThemeWin::supportsFocus const):
(WebCore::RenderThemeWin::determineClassicState):
(WebCore::RenderThemeWin::determineState):
(WebCore::RenderThemeWin::getClassicThemeData):
(WebCore::RenderThemeWin::getThemeData):
(WebCore::drawControl):
(WebCore::RenderThemeWin::paintSliderTrack):
(WebCore::RenderThemeWin::adjustSliderThumbSize const):
(WebCore::RenderThemeWin::supportsMeter const):
* Source/WebCore/rendering/RenderThemeWin.h:
* Source/WebCore/rendering/style/RenderStyle.h:
(WebCore::RenderStyle::hasAppearance const):
(WebCore::RenderStyle::hasEffectiveAppearance const):
(WebCore::RenderStyle::appearance const):
(WebCore::RenderStyle::effectiveAppearance const):
(WebCore::RenderStyle::setAppearance):
(WebCore::RenderStyle::setEffectiveAppearance):
(WebCore::RenderStyle::initialAppearance):

Canonical link: <a href="https://commits.webkit.org/257791@main">https://commits.webkit.org/257791@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c7b65ecb810d67efa43dc1a0cc2cae1acec63d36

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/99987 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/9156 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/33067 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/109334 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/169567 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/103986 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/10040 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/86457 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/92436 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/107234 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/105755 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/7587 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/90868 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/34305 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/89493 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/22259 "Passed tests") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/2951 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/23772 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/2911 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/46145 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/9044 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/43250 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5349 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/4759 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebCore/WebCore.xcodeproj/project.pbxproj:
```diff
@@ -25802,6 +25805,15 @@
+				722728122947A4090048DD2E /* ControlPartType.cpp */,
+				722728112947A18A0048DD2E /* ControlPartType.h */,
+			);
+			path = formcontrols;
```
Nit: I would name the folder `form-controls`.

shallawa replied:
I agree but there are many precedents which do not use dashes or upper-case letters to make the folder name clearer:

```
contentextensions
platform/encryptedmedia
platform/graphics/coreimage
platform/graphics/coretext
platform/graphics/displaylists
platform/graphics/opentype
platform/mediacapabilities
platform/mediarecorder
...
```
I do not know the rationale behind this naming convention. The only exception I saw was `platform/image-decoders`.

shallawa committed:
[GPU Process] [FormControls] Rename ControlPart to ControlPartType and make it enum class
https://bugs.webkit.org/show_bug.cgi?id=249167
rdar://103266751

Reviewed by Aditya Keerthi.

The plan is to add a new class called ControlPart in which arguments from the DOM
elements and the render elements will be passed to GraphicsContext. GraphicsContext
will be responsible of drawing the ControlPart so we can delegate the drawing as
a DisplayList item to GPUP.

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
(WebCore::AccessibilityRenderObject::isApplePayButton const):
* Source/WebCore/accessibility/AccessibilitySlider.cpp:
(WebCore::AccessibilitySlider::orientation const):
* Source/WebCore/css/CSSPrimitiveValueMappings.h:
(WebCore::CSSPrimitiveValue::CSSPrimitiveValue):
(WebCore::CSSPrimitiveValue::operator ControlPartType const):
(WebCore::CSSPrimitiveValue::operator ControlPart const): Deleted.
* Source/WebCore/css/CSSValueKeywords.in:
* Source/WebCore/html/RangeInputType.cpp:
(WebCore::RangeInputType::handleKeydownEvent):
* Source/WebCore/html/shadow/SliderThumbElement.cpp:
(WebCore::hasVerticalAppearance):
(WebCore::SliderThumbElement::resolveCustomStyle):
* Source/WebCore/html/shadow/TextControlInnerElements.cpp:
(WebCore::SearchFieldResultsButtonElement::resolveCustomStyle):
(WebCore::SearchFieldCancelButtonElement::resolveCustomStyle):
* Source/WebCore/platform/Theme.cpp:
(WebCore::Theme::baselinePositionAdjustment const):
(WebCore::Theme::controlFont const):
(WebCore::Theme::controlSize const):
(WebCore::Theme::minimumControlSize const):
(WebCore::Theme::controlRequiresPreWhiteSpace const):
(WebCore::Theme::paint):
(WebCore::Theme::inflateControlPaintRect const):
(WebCore::Theme::controlBorder const):
(WebCore::Theme::controlPadding const):
* Source/WebCore/platform/Theme.h:
* Source/WebCore/platform/ThemeTypes.cpp:
* Source/WebCore/platform/ThemeTypes.h:
* Source/WebCore/platform/adwaita/ThemeAdwaita.cpp:
(WebCore::ThemeAdwaita::controlSize const):
(WebCore::ThemeAdwaita::minimumControlSize const):
(WebCore::ThemeAdwaita::controlBorder const):
(WebCore::ThemeAdwaita::paint):
* Source/WebCore/platform/adwaita/ThemeAdwaita.h:
* Source/WebCore/platform/graphics/formcontrols/ControlPartType.cpp: Added.
(WebCore::operator<<):
* Source/WebCore/platform/graphics/formcontrols/ControlPartType.h: Added.
* Source/WebCore/platform/mac/ThemeMac.h:
* Source/WebCore/platform/mac/ThemeMac.mm:
(WebCore::configureToggleButton):
(WebCore::createToggleButtonCell):
(WebCore::sharedRadioCell):
(WebCore::sharedCheckboxCell):
(WebCore::paintToggleButton):
(WebCore::setUpButtonCell):
(WebCore::button):
(WebCore::paintButton):
(WebCore::paintColorWell):
(WebCore::ThemeMac::baselinePositionAdjustment const):
(WebCore::ThemeMac::controlFont const):
(WebCore::ThemeMac::controlSize const):
(WebCore::ThemeMac::minimumControlSize const):
(WebCore::ThemeMac::controlBorder const):
(WebCore::ThemeMac::controlPadding const):
(WebCore::ThemeMac::inflateControlPaintRect const):
(WebCore::ThemeMac::paint):
* Source/WebCore/rendering/RenderAttachment.cpp:
(WebCore::RenderAttachment::shouldDrawBorder const):
* Source/WebCore/rendering/RenderLayerScrollableArea.cpp:
(WebCore::RenderLayerScrollableArea::updateScrollbarsAfterStyleChange):
(WebCore::RenderLayerScrollableArea::updateScrollbarsAfterLayout):
* Source/WebCore/rendering/RenderListBox.cpp:
(WebCore::RenderListBox::createScrollbar):
* Source/WebCore/rendering/RenderMenuList.cpp:
(RenderMenuList::menuStyle const):
(RenderMenuList::clientPaddingLeft const):
(RenderMenuList::clientPaddingRight const):
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::adjustAppearanceForElement const):
(WebCore::isAppearanceAllowedForAllElements):
(WebCore::RenderTheme::adjustStyle):
(WebCore::RenderTheme::autoAppearanceForElement const):
(WebCore::RenderTheme::paint):
(WebCore::RenderTheme::paintBorderOnly):
(WebCore::RenderTheme::paintDecorations):
(WebCore::RenderTheme::isControlContainer const):
(WebCore::RenderTheme::isControlStyled const):
(WebCore::RenderTheme::supportsFocusRing const):
(WebCore::RenderTheme::isDefault const):
(WebCore::RenderTheme::supportsMeter const):
(WebCore::RenderTheme::paintSliderTicks):
* Source/WebCore/rendering/RenderTheme.h:
(WebCore::RenderTheme::scrollbarControlSizeForPart):
(WebCore::RenderTheme::canPaint const):
* Source/WebCore/rendering/RenderThemeAdwaita.cpp:
(WebCore::RenderThemeAdwaita::supportsFocusRing const):
(WebCore::RenderThemeAdwaita::popupInternalPaddingBox const):
(WebCore::RenderThemeAdwaita::paintMenuList):
(WebCore::RenderThemeAdwaita::paintSliderTrack):
(WebCore::RenderThemeAdwaita::adjustSliderThumbSize const):
(WebCore::RenderThemeAdwaita::paintSliderThumb):
* Source/WebCore/rendering/RenderThemeIOS.h:
* Source/WebCore/rendering/RenderThemeIOS.mm:
(WebCore::RenderThemeIOS::canPaint const):
(WebCore::RenderThemeIOS::adjustedPaintRect const):
(WebCore::RenderThemeIOS::baselinePosition const):
(WebCore::RenderThemeIOS::isControlStyled const):
(WebCore::RenderThemeIOS::popupInternalPaddingBox const):
(WebCore::canAdjustBorderRadiusForAppearance):
(WebCore::RenderThemeIOS::paintSliderTrack):
(WebCore::RenderThemeIOS::adjustSliderThumbSize const):
(WebCore::RenderThemeIOS::adjustButtonStyle const):
(WebCore::RenderThemeIOS::supportsBoxShadow const):
(WebCore::RenderThemeIOS::supportsMeter const):
(WebCore::RenderThemeIOS::paintSliderTicks):
(WebCore::RenderThemeIOS::paintSliderTrackWithFormControlRefresh):
* Source/WebCore/rendering/RenderThemeMac.h:
* Source/WebCore/rendering/RenderThemeMac.mm:
(WebCore::RenderThemeMac::canPaint const):
(WebCore::RenderThemeMac::isControlStyled const):
(WebCore::RenderThemeMac::adjustRepaintRect):
(WebCore::RenderThemeMac::controlSupportsTints const):
(WebCore::RenderThemeMac::meterSizeForBounds const):
(WebCore::RenderThemeMac::supportsMeter const):
(WebCore::RenderThemeMac::levelIndicatorFor const):
(WebCore::RenderThemeMac::progressBarRectForBounds const):
(WebCore::RenderThemeMac::popupInternalPaddingBox const):
(WebCore::RenderThemeMac::paintSliderTrack):
(WebCore::RenderThemeMac::paintSliderThumb):
(WebCore::RenderThemeMac::adjustSliderThumbSize const):
* Source/WebCore/rendering/RenderThemeWin.cpp:
(WebCore::RenderThemeWin::supportsFocus const):
(WebCore::RenderThemeWin::determineClassicState):
(WebCore::RenderThemeWin::determineState):
(WebCore::RenderThemeWin::getClassicThemeData):
(WebCore::RenderThemeWin::getThemeData):
(WebCore::drawControl):
(WebCore::RenderThemeWin::paintSliderTrack):
(WebCore::RenderThemeWin::adjustSliderThumbSize const):
(WebCore::RenderThemeWin::supportsMeter const):
* Source/WebCore/rendering/RenderThemeWin.h:
* Source/WebCore/rendering/style/RenderStyle.h:
(WebCore::RenderStyle::hasAppearance const):
(WebCore::RenderStyle::hasEffectiveAppearance const):
(WebCore::RenderStyle::appearance const):
(WebCore::RenderStyle::effectiveAppearance const):
(WebCore::RenderStyle::setAppearance):
(WebCore::RenderStyle::setEffectiveAppearance):
(WebCore::RenderStyle::initialAppearance):

Canonical link: https://commits.webkit.org/257791@main
